### PR TITLE
refactor: Optimized some functionalities

### PR DESCRIPTION
**Pull request by kaushikb11**
Worked on the redundant lines of code. Working on the new concepts of Pacman canvas to make it efficient.

### CODE REVIEW

File: client/src/components/PacmanGame/PacmanBoard.jsx [hunk 1]
```diff
@@ -14,9 +13,17 @@ function PacmanBoard({ gridState, gridSize }) {
   if (gridState === '') {
     return null;
   }
-  console.log(gridSize);
-  console.log(codeToEntity(gridState[43][5]));
   let game = [];
+
+  const flattenArray = (inputArray) => {
```
Comment (Shalinit3) [inline client/src/components/PacmanGame/PacmanBoard.jsx:18]: You can use Array.prototype.flat() instead.

File: client/src/components/PacmanGame/entities/Pacman.jsx [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
+      direction: KEYS.right,
+      x: 0,
+      y: 0
+    }
+  }
+
+  render () {
+    return (
+           <Rect
+                x={this.props.x * this.props.gridSize}
```
Comment (Shalinit3) [inline client/src/components/PacmanGame/entities/Pacman.jsx:19]: Destructure the props.

File: client/src/components/PacmanGame/PacmanBoard.jsx [hunk 3]
```diff
@@ -14,9 +13,17 @@ function PacmanBoard({ gridState, gridSize }) {
...
+    const flattenedArray = [];
+    for (let i=0; i < inputArray.length; ++i) {
+        for (let j=0; j< inputArray[i].length; ++j)
+            flattenedArray.push(inputArray[i][j]);
+    return flattenedArray
+  }
+  }
+
   for (let row = 0; row < gridState.length; row++) {
```
Comment (Shalinit3) [inline client/src/components/PacmanGame/PacmanBoard.jsx:27]: This part can be refactored.

**Comment by Shalinit3** [comment]
Name of the PR is the last commit message.

File: client/src/components/PacmanGame/PacmanBoard.jsx (hunk 1 above)
Comment (kaushikb11) [reply client/src/components/PacmanGame/PacmanBoard.jsx:18]: https://jsperf.com/flatten-an-array-loop-vs-reduce/38
 Loop works better than _reduce_ and _prototype.flat()_ while flattening the array.

File: client/src/components/PacmanGame/entities/Pacman.jsx (hunk 2 above)
Comment (kaushikb11) [reply client/src/components/PacmanGame/entities/Pacman.jsx:19]: Pacman.js is still WIP. Will refactor it in the next commit with event listeners and few methods

File: client/src/components/PacmanGame/PacmanBoard.jsx (hunk 1 above)
Comment (Shalinit3) [reply client/src/components/PacmanGame/PacmanBoard.jsx:18]: Array.flat() completed first.

Comment (kaushikb11) [reply client/src/components/PacmanGame/PacmanBoard.jsx:18]: Read the x-axis. It's the operations per second.

File: client/src/components/PacmanGame/entities/Pacman.jsx (hunk 2 above)
Comment (Shalinit3) [reply client/src/components/PacmanGame/entities/Pacman.jsx:19]: Why are we merging it then. We can complete this part and then merge.

File: client/src/components/PacmanGame/entities/Pacman.jsx [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
-    y={y * gridSize}
-    width={gridSize}
-    height={gridSize}
-    fill="yellow"
-  />
-);
+const KEYS = {left: 37, up: 38, right: 39, down: 40};
+
+class Pacman extends React.Component {
+  constructor(props) {
```
Comment (shivangg) [inline client/src/components/PacmanGame/entities/Pacman.jsx:7]: use class variables instead of constructor

**Review by shivangg — CHANGES_REQUESTED** [review]

**Pull request closed without merging**